src/v/cloud_storage/cache_service.cc

```diff
@@ -529,6 +529,8 @@ ss::future<cache::trim_result> cache::trim_fast(
     trim_result result;
 
     size_t candidate_i = 0;
+    const std::regex segment_expr{
+      R"#(.*\.log(\.\d+)?)#", std::regex_constants::optimize};
```

Comment:
Could we avoid constructing this regex on every invocation to `trim_fast` (e.g. make it a member)?

Reply:
i think we should also use re2, since it's already linked

Reply:
Moved to anonymous namespace